test/test_xmath_result_type.cpp

```diff
@@ -66,21 +66,32 @@ namespace xt
         CHECK_RESULT_TYPE(FUNC<signed int>(INPUT), signed int);                      \
         CHECK_RESULT_TYPE(FUNC<int>(INPUT), int);                                    \
         CHECK_RESULT_TYPE(FUNC<unsigned long long>(INPUT), unsigned long long);      \
-        CHECK_RESULT_TYPE(FUNC<signed long long>(INPUT), signed long long);
+        CHECK_RESULT_TYPE(FUNC<signed long long>(INPUT), signed long long);          \
+        CHECK_RESULT_TYPE(FUNC<long long>(INPUT), long long);                        \
+        CHECK_RESULT_TYPE(FUNC<float>(INPUT), float);                                \
+        CHECK_RESULT_TYPE(FUNC<double>(INPUT), double);
 
     TEST(xmath, result_type)
     {
         shape_type shape = {3, 2};
         xarray<unsigned char> auchar(shape);
         xarray<short> ashort(shape);
+        xarray<unsigned short> aushort(shape);
         xarray<int> aint(shape);
         xarray<unsigned int> auint(shape);
+        xarray<long long> along(shape);
         xarray<unsigned long long> aulong(shape);
         xarray<float> afloat(shape);
         xarray<double> adouble(shape);
         xarray<std::complex<float>> afcomplex(shape);
         xarray<std::complex<double>> adcomplex(shape);
 
+#define CHECK_TEMPLATED_RESULT_TYPE_FOR_ALL(INPUT)                                   \
+        CHECK_TEMPLATED_RESULT_TYPE(mean, INPUT)                                     \
+        CHECK_TEMPLATED_RESULT_TYPE(variance, INPUT)
+// FIXME: the first 6 checks in "#define CHECK_TEMPLATED_RESULT_TYPE(FUNC, INPUT)" fail
+//        CHECK_TEMPLATED_RESULT_TYPE(stddev, INPUT)
```

Comment:
The `stddev` test fails because

```
CHECK_TEMPLATED_RESULT_TYPE(stddev, INPUT)
```
fails with the following tests
```
        CHECK_RESULT_TYPE(FUNC<unsigned char>(INPUT), int);                          \
        CHECK_RESULT_TYPE(FUNC<signed char>(INPUT), int);                            \
        CHECK_RESULT_TYPE(FUNC<char>(INPUT), int);                                   \
        CHECK_RESULT_TYPE(FUNC<unsigned short>(INPUT), int);                         \
        CHECK_RESULT_TYPE(FUNC<signed short>(INPUT), int);                           \
        CHECK_RESULT_TYPE(FUNC<short>(INPUT), int);                                  \
```
I guess it is because of the implicit conversion in `xt::mean` while the result type of `xt::stddev` is obtained by `xt::cast`?

Reply:
Indeed.